docs/pairing.md

```diff
@@ -0,0 +1,114 @@
+    "reported": {
+      "thief": {
+        "pairing": {
+          "deviceRunId": "4d41c744-94bf-40ac-89bc-06f28b4dc9d2",
+          "pairingId": "176e7469-9755-4a7a-af74-85f735f0f296",
+          "requestedServicePool": "bertk_desktop_pool",
+          "serviceRunId": null
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `deviceRunId` | guid | `runId` for the running device app instance.  re-generated each time the app launches |
+| `pairingId` | guid | ID representing an individual pairing attempt.  The device generates a new pairingId each time it tries to pair (or re-pair) with a service app |
+| `requestedServicePool` | string | free-form name for the pool of service apps which are known to be valid.  This is the the only value that a service app uses to decide if it can pair with the device app |
+| `serviceRunId` | guid | runId of the selected service app. Since this step is starting the pairing process, this is set to `null` because no service app has been chosen yet. |
+
+## Step 2: service sets desired properties to say that it's available.
+
+A service app can tell the device app that it's available for pairing by setting `properties/desired/thief/pairing` values as described below.
+
+```json
+  {
+    "desired": {
+      "thief": {
+        "pairing": {
+          "pairingId": "176e7469-9755-4a7a-af74-85f735f0f296",
+          "serviceRunId": "23ebf618-41e2-40d7-9964-a16ae9762a1c",
+          "acceptedPairing": null
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `pairingId` | guid | ID representing an individual pairing attempt.  This is included so the device can know that it's not looking at an old set of desired properties. |
+| `serviceRunId` | guid | `runId` for the service app that wants to pair with the device app |
+| `acceptedPairing` | guid | `none` in this step because there is no agreement (yet) between the device app and service app |
+
+## Step 3: device sets reported properties to select service instance.
+
+The device app selects a service instance by setting `properties/reported/thief/pairing/serviceRunId` to the service app's runId value.
+
+```json
+  {
+    "reported": {
+      "thief": {
+        "pairing": {
+          "pairingId": "176e7469-9755-4a7a-af74-85f735f0f296",
+          "serviceRunId": "23ebf618-41e2-40d7-9964-a16ae9762a1c"
+        }
+      }
+    }
+  }
+```
+
+| field | format | meaning |
+| - | - | - |
+| `pairingId` | guid | ID representing an individual pairing attempt.  Used like a "transaction ID" so the service can know that the device is still working on the same pairing operation |
+| `serviceRunId` | guid | `runId` for the service app that was selected by the device app |
+
+## Step 4: service sets desired properties to acknowledge that the pairing is complete.
+
+When the service app sees that it has been selected,  it sets `properties/desired/thief/pairing/acceptedPairing` to "{pairingId},{serviceRunId}".
```

Comment:
Isn't this a duplicate of two other fields on the twin?

I've gotten confused about the process. It might be slightly more complicated than it needs to be. Perhaps we should talk through it and see if we can simplify it a tad.

Reply:
I think we can.  I think I can make it work without steps 3 and 4 if I make a few (hopefully safe) assumptions.